src/hash_builder/mod.rs

```diff
@@ -200,8 +200,18 @@ impl HashBuilder {
             let preceding_len = self.groups.len().saturating_sub(1);
 
             let common_prefix_len = succeeding.common_prefix_length(current.as_slice());
-            let len = cmp::max(preceding_len, common_prefix_len);
-            assert!(len < current.len(), "len {} current.len {}", len, current.len());
+
+            // guard against accessing beyond the current key length
+            if common_prefix_len >= current.len() {
+                return;
+            }
```

Comment:
`common_prefix_len > current.len()` is just not possible here due to `common_prefix_length` call. that leaves us with `common_prefix_len == current.len()`, why the change of behavior to return early